### ISSUE TITLE
Algolia Search PHP Client v4 is not Compatible

### ISSUE THREAD

**Issue by Gautamsam3** [laravel/scout#855]
### Scout Version

10.11.1

### Scout Driver

Algolia

### Laravel Version

11

### PHP Version

8.3.8

### Description

As per Laravel's [documentation](https://laravel.com/docs/11.x/scout#algolia), user needs to install Alogila client via following command:
 ```
 composer require algolia/algoliasearch-client-php`
 ```
 This by default installs latest version v4 of the Algolia Search client, which was recently [released](https://github.com/algolia/algoliasearch-client-php/releases).

![algolia_v41](https://github.com/user-attachments/assets/457e344d-4c7f-4bbc-b0cf-f63238772a69)

## Issue:
Despite the successful installation, `php artisan scout:import "App\Models\Variant"` command  results an error.

![algolia_v4](https://github.com/user-attachments/assets/515aaa3d-ed37-474b-ac77-ec1b00dbd4d5)

### Steps To Reproduce

1. Install `algolia/algoliasearch-client-php` in the project.
2. [Configure ](https://laravel.com/docs/11.x/scout#configuration)model.
3. Import `php artisan scout:import "App\Models\<model name>"`

**Comment by driesvints** [comment]
Thanks. Welcoming PR's to make sure we're compatible with v4 👍

**Comment by alexander-schranz** [comment]
If somebody require some references I did work on v3 -> v4 update in my own package. https://github.com/schranz-search/schranz-search/pull/414/files.

Linking also the Upgrade Guide from Algolia: https://www.algolia.com/doc/libraries/php/v4/upgrade/

The hard part is mostly if you want to keep V3 support then I think its lot of more work, else you can only do it for next major.

**Comment by driesvints** [comment]
We can cut a new major release if need be.

**Comment by karandatwani92** [comment]
Thanks @alexander-schranz 

Your links helped a lot. I think the scout repo doesn't need many changes. I have made a few changes locally. Things worked fine for now. I'll be testing a little more tomorrow before submitting the PR

I hope everything goes well. I'm very excited to submit my first PR on Laravel's repo.

**Comment by muhghazaliakbar** [comment]
Is there any update on this?

**Comment by thumbtech** [comment]
> We can cut a new major release if need be.

Yes this should/could have been a major release as it does break algolia/scout-extended dependencies

https://github.com/algolia/scout-extended/issues/347

**Comment by mariuscucuruz** [comment]
I've submitted PR https://github.com/algolia/scout-extended/pull/348 to downgrade [Laravel Scout to v10.11.5](https://packagist.org/packages/laravel/scout#v10.11.5).

While we await approval and merge, you can run this Composer command in your project(s):
```
composer update laravel/scout v10.11.5
```

---

> Thank you for reporting this issue!
> 
> As Laravel is an open source project, we rely on the community to help us diagnose and fix issues as it is not possible to research and fix every issue reported to us via GitHub.
> 
> If possible, please make a pull request fixing the issue you have described, along with corresponding tests. All pull requests are promptly reviewed by the Laravel team.
> 
> Thank you!

**Comment by mariuscucuruz** [comment]
> Is there any update on this?

Hey [ThumbTech](https://github.com/thumbtech),

You can to downgrade [Laravel Scout to v10.11.5](https://packagist.org/packages/laravel/scout#v10.11.5) by running this Composer command in your project(s):
```
composer update laravel/scout v10.11.5
```

**Comment by thumbtech** [comment]
> composer update laravel/scout v10.11.5

`composer require laravel/scout v10.11.5` actually does work

(Better IMO to leave scout-extended as is and just commit to v11 on scout, as numerous others have recommended.)

**Comment by mariuscucuruz** [comment]
this compatibility issue has been addressed in [#348](https://github.com/algolia/scout-extended/pull/348) and subsequently with the release of [v3.1.1.](https://packagist.org/packages/algolia/scout-extended#v3.1.1).
